SillyBusinessInc/Moldbreaker
# Origin/feat/#262 breaking platform

## Purpose of this PR:
This is a script and it should be placable on any Object. 

logic
- if the player collides, after few seconds the platform will start cracking. 
- After cracking the platform will break and disappear.
- after a few seconds of delay, it respawns again in the same position.

in the inspector 
- delay : seconds between collision and start cracking.
- respawn delay : seconds between start cracking and respawning.(its not between destroy and respawn)
- can respawn : checked - platform respawns / unchecked - platform is disposable
- can break : checked - platform will break / unchecked - platform will not break

## How to test:
you can attach breakingPlatformController to an object.(the object must contain Animator, Collider, Renderer)
On the Animator the default Controller is BreakingPlatform. there is 4 stat(Idle, cracking, breaking, broken). you can add animation in cracking and breaking. for now there's no animation attached. So if you want to check how it's working, you can check the animation loop in the Animator. 

### links: 
https://github.com/SillyBusinessInc/Moldbreaker/issues/262
